scripts/chandra/download_script.py

```diff
@@ -0,0 +1,160 @@
+# Astropile - Rafael Martinez-Galarza & Mike Tibbetts (CfA)
+# This code dowloads data files from the Chandra Source Catalog
+# using the CLI protocol:
+# https://cxc.cfa.harvard.edu/ciao/threads/csccli/
+# For each X-ray detection (each one associated to a source name 
+# and RA and DEC, it brings down the folowing files: 
+# 1) Event file - this is the file containing all photon recordings
+# 2) Light curve - binned light curves from Gregory-Loredo alg.
+# 3) Spectral files - PHA (signal), and ARF and RMF (response)
+# 4) Images - fits files containing the counts binned spatially
+# 5) PSF - the computed PSF at the position of the detection
+# The data downloaded correspond to X-ray sources with enough
+# counts to allow for spectral fitting (and a minimum of 50 counts),
+# that have S/N of at least 5, and that are at most 5 arcmin off-axis
+# in the field of view. Also, all events in the broadband (0.5keV-7.0keV)
+# are included
+
+
+# First, let's get the catalog data, directly from the release
+# To donwload the catalog table that associates X-ray sources 
+# (and sky coordinates) to each detection, and therefore to
+# the data products, use the following:
+
+# Imports
+import pyvo as vo
+import requests
+import argparse
+import h5py
+import numpy as np
+
+# CSC 2.1 TAP service
+tap = vo.dal.TAPService('http://cda.cfa.harvard.edu/csc21tap') # For CSC 2.1
+
+
+def get_source_detections_ids(args):
+
+    # Define the minimum source counts, minimum significance, and output file
+    # Recommend: min_cnts = 4000, min_sig = 40, max_theta = 1
+    # This functon will create a list of  IDs
+    min_cnts, min_sig, max_theta, output_file,file_path = args
+
+
+    qry = """
+    SELECT m.name, m.ra, m.dec, o.obsid, o.obi, o.region_id, o.src_cnts_aper_b,
+    o.flux_significance_b, o.flux_aper_b, o.theta, o.flux_bb_aper_b,
+    o.gti_mjd_obs, o.hard_hm,o.hard_hs, o.hard_ms, o.var_prob_b, 
+    o.var_index_b 
+    FROM csc21.master_source m, csc21.master_stack_assoc a, csc21.observation_source o, 
+    csc21.stack_observation_assoc b, csc21.stack_source s 
+    WHERE ((a.match_type = 'u') AND (o.flux_bb_aper_b IS NOT NULL) 
+    AND (o.src_cnts_aper_b > """+str(min_cnts)+""") AND (o.flux_significance_b > """+str(min_sig)+""") 
+    AND (o.theta < """+str(max_theta)+""")) AND (m.name = a.name) 
+    AND (s.detect_stack_id = a.detect_stack_id and s.region_id = a.region_id) 
+    AND (s.detect_stack_id = b.detect_stack_id and s.region_id = b.region_id) 
+    AND (o.obsid = b.obsid and o.obi = b.obi and o.region_id = b.region_id)
+    ORDER BY name ASC
+    """
+
+    cat = tap.search(qry)
+
+    # Convert the catalog to an astropy Table format.
+    cat = cat.to_table()
+
+    # Save the catalog to HDF5 format
+    with h5py.File(file_path+output_file+'_hdf5.hdf5', 'w') as hdf5_file:
```

Comment:
```suggestion
    with h5py.File(file_path+output_file+'.hdf5', 'w') as hdf5_file:
```
This `_hdf5` name in the file name is superfluous given the extension.